pkg/expr/sql/parser_allow.go

```diff
@@ -54,7 +54,13 @@ func allowedNode(node sqlparser.SQLNode) (b bool) {
 	case *sqlparser.AliasedExpr, *sqlparser.AliasedTableExpr:
 		return
 
-	case *sqlparser.BinaryExpr:
+	case *sqlparser.AndExpr, *sqlparser.OrExpr:
```

Comment:
are we primarily concerned with functions? if so, i am wondering if implementing `FunctionProvider` and moving the allow list there would make this easier to maintain

Reply:
No, also Nodes (e.g. `EXECUTE` I think). The main change was to move to an allow list, so we are always checking what is allowed through.

There still might be non-parser way to do this with the engine though? Having it both places would feel more secure (defense in depth).

Reply:
> Having it both places would feel more secure

yeah, a bit of extra maintenance i guess, but everything helps